x-pack/test/functional/apps/api_keys/home_page.ts

```diff
@@ -10,18 +10,33 @@ import { FtrProviderContext } from '../../ftr_provider_context';
 export default ({ getPageObjects, getService }: FtrProviderContext) => {
   const pageObjects = getPageObjects(['common', 'apiKeys']);
   const log = getService('log');
+  const security = getService('security');
+  const testSubjects = getService('testSubjects');
 
   describe('Home page', function() {
     this.tags('smoke');
     before(async () => {
+      await security.testUser.setRoles(['kibana_admin']);
       await pageObjects.common.navigateToApp('apiKeys');
     });
 
+    after(async () => {
+      await security.testUser.restoreDefaults();
+    });
+
+    // https://www.elastic.co/guide/en/kibana/7.6/api-keys.html#api-keys-security-privileges
+    it('Shows required privileges ', async () => {
+      log.debug('Checking for section header');
+      const messageElement = await testSubjects.find('apiKeysPermissionDeniedMessage');
+      const message = await messageElement.getVisibleText();
+      expect(message).to.be('You need permission to manage API keys');
+    });
+
     it('Loads the app', async () => {
+      await security.testUser.setRoles(['kibana_admin', 'test_api_keys']);
       log.debug('Checking for section header');
       const headerText = await (await pageObjects.apiKeys.noAPIKeysHeading()).getVisibleText();
       expect(headerText).to.be('No API keys');
```

Comment:
I know this wasn't part of this PR, but if you could also change `pageObjects.apiKeys.noAPIKeysHeading()` to return the visible text, that would clean up the test a bit.

Reply:
adopted the suggestion